modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxStateImpl.java

```diff
@@ -222,11 +222,7 @@ public class IgniteTxStateImpl extends IgniteTxLocalStateAdapter {
 
         this.recovery = recovery;
 
-        if (mvccEnabled != null && mvccEnabled != cacheCtx.mvccEnabled())
-            throw new IgniteCheckedException("Failed to enlist new cache to existing transaction " +
-                "(caches with different mvcc settings can't be enlisted in one transaction).");
-
-        mvccEnabled = cacheCtx.mvccEnabled();
+        mvccEnabled = false;
```

Comment:
I think we can remove IgniteTxStateImpl#mvccEnabled. WDYT?

Reply:
Sure, but later. 
Trying to keep PR changes size reasonable